docs/tutorials/ptypy-workflows.md

```diff
@@ -0,0 +1,440 @@
+the ```sessionspaces``` configMap to describe the path to the visit directory in /dls.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ClusterWorkflowTemplate
+metadata:
+  name: ptypy-cpu-job
+spec:
+  entrypoint: ptypy-run
+  arguments:
+    parameters:
+    - name: visitdir
+      valueFrom:
+        configMapKeyRef:
+          name: sessionspaces
+          key: data_directory
+  templates:
+  - name: ptypy-run
+    container:
+      volumeMounts:
+      - name: session
+        mountPath: "{{ workflow.parameters.visitdir }}"
+    volumes:
+    - name: session
+      hostPath:
+        path:  "{{ workflow.parameters.visitdir }}"
+        type: Directory
+```
+
+### CPU cluster resources
+
+By default, each pod on the cluster is allocating a single CPU and 1 GB of memory.
+If we would like to specify different resources, we can do so by providing a ```podSpecPatch```
+to the main workflow pod
+
+```yaml
+
+  templates:
+  - name: ptypy-run
+    inputs:
+      parameters:
+      - name: nprocs
+        value: 1
+      - name: memory
+        value: 20Gi
+    podSpecPatch: |
+      containers:
+        - name: main
+          resources:
+            requests:
+              cpu: "{{ inputs.parameters.nprocs }}"
+              memory: "{{ inputs.parameters.memory }}"
+            limits:
+              cpu: "{{ inputs.parameters.nprocs }}"
+              memory: "{{ inputs.parameters.memory }}"
+```
+
+and use input parameters for the number of CPUs (```nprocs```) and the requested memory (```memory```)
+like described above.
+
+### GPU resources
+
+For GPU jobs, we need to add entries for  ```nvidia.com/gpu``` in both the resource ```requests``` and 
+```limits```. In addition, we also need to add some ```tolerations```:
+
+```yaml
+  templates:
+  - name: ptypy-run
+    inputs:
+      parameters:
+      - name: nprocs
+        value: 1
+      - name: memory
+        value: 20Gi
+    podSpecPatch: |
+      containers:
+        - name: main
+          resources:
+            requests:
+              cpu: "{{ inputs.parameters.nprocs }}"
+              memory: "{{ inputs.parameters.memory }}"
+              nvidia.com/gpu: "{{ inputs.parameters.nprocs }}"
+            limits:
+              cpu: "{{ inputs.parameters.nprocs }}"
+              memory: "{{ inputs.parameters.memory }}"
+              nvidia.com/gpu: "{{ inputs.parameters.nprocs }}"
+    tolerations:
+    - key: nvidia.com/gpu
+      operator: Exists
+      effect: NoSchedule
+    - key: nodetype
+      operator: Equal
+      value: gpu
+      effect: NoSchedule
+```
+
+### Non-root container execution
+
+For security reasons, workflow pods are only allowed to run with non-root users,
+this means that the application running inside the container won't have write access
+to anything except mounted directories.
+Since many applications will need to write to some locations such as ```/tmp```,
+it makes sense to add a volume claim for a temporary directory to our template
```

Comment:
Probably better to phrase this as "if required" - a lot of containers won't need this

Reply:
Done a bit of rewriting to make clear that this is optional and might not be needed.